src/visitors/sympy_replace_solutions_visitor.cpp

```diff
@@ -0,0 +1,350 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include "visitors/sympy_replace_solutions_visitor.hpp"
+#include "visitors/lookup_visitor.hpp"
+
+
+#include "ast/all.hpp"
+#include "utils/logger.hpp"
+#include "visitors/visitor_utils.hpp"
+
+namespace nmodl {
+namespace visitor {
+
+
+void SympyReplaceSolutionsVisitor::InterleavesCounter::new_equation(const bool is_in_system) {
+    n_ += (!in_system_ && is_in_system);
```

Comment:
I improved the docs. It is a +1 in case `in_system_ == false` and `is_in_system == true` (a switch). Maybe I can rename `in_system_` -> `old_in_system_` to stress out that it follows `is_in_system`

Reply:
improved the docs